app/src/main/java/tech/pauly/findapet/data/FilterRepository.java

```diff
@@ -0,0 +1,48 @@
+package tech.pauly.findapet.data;
+
+import android.support.annotation.VisibleForTesting;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import tech.pauly.findapet.data.models.Filter;
+
+@Singleton
+public class FilterRepository {
+
+    private static long ABSENT = -1L;
+
+    private FilterDatabase database;
+    private ObservableHelper observableHelper;
+
+    @VisibleForTesting
+    Long currentFilterId = ABSENT;
```

Comment:
Not a super huge fan of keeping state here in the repo. However, didn't want to overcomplicate it at the moment, since I'm going to have to keep multiple current filters and everything soon enough, just thought to KISS for now.